AwsCryptographicMaterialProviders/runtimes/java/src/test/LocalCMCTests.java

```diff
@@ -79,37 +79,39 @@ public void TestALotOfAdding() {
       .identifier(cacheIdentifier)
       .build();
 
-    try {
-      GetCacheEntryOutput getCacheEntryOutput = test.GetCacheEntry(
-        getCacheEntryInput
-      );
-      //      assertEquals(getCacheEntryOutput.materials().BeaconKey().beaconKey(), binaryData);
-      //      assertEquals(getCacheEntryOutput.materials().BeaconKey().beaconKeyIdentifier(),
-      // stringData);
-      //      System.out.println("are equal");
-    } catch (EntryDoesNotExist ex) {
-      Materials materials = Materials
-        .builder()
-        .BeaconKey(
-          BeaconKeyMaterials
-            .builder()
-            .beaconKeyIdentifier(beaconKeyIdentifier)
-            // The cacheIdentifier is used as the material
-            // because we are not testing the cryptography here.
-            .beaconKey(cacheIdentifier)
-            .encryptionContext(new HashMap<String, String>())
-            .build()
-        )
-        .build();
+    for (int i = 0; i < 100; i++) {
+      try {
+        GetCacheEntryOutput getCacheEntryOutput = test.GetCacheEntry(
+          getCacheEntryInput
+        );
+      } catch (InterruptedException ex) {
+        String str = String.format("Caught InterruptedException %d", i);
+        System.err.println(str);
+        continue;
+      } catch (EntryDoesNotExist ex) {
+        Materials materials = Materials
+          .builder()
+          .BeaconKey(
+            BeaconKeyMaterials
+              .builder()
+              .beaconKeyIdentifier(beaconKeyIdentifier)
+              // The cacheIdentifier is used as the material
+              // because we are not testing the cryptography here.
+              .beaconKey(cacheIdentifier)
+              .encryptionContext(new HashMap<String, String>())
+              .build()
+          )
+          .build();
 
-      PutCacheEntryInput putCacheEntryInput = PutCacheEntryInput
-        .builder()
-        .identifier(cacheIdentifier)
-        .creationTime(Instant.now().getEpochSecond())
-        .expiryTime(Instant.now().getEpochSecond() + 1)
-        .materials(materials)
-        .build();
-      test.PutCacheEntry(putCacheEntryInput);
+        PutCacheEntryInput putCacheEntryInput = PutCacheEntryInput
+          .builder()
+          .identifier(cacheIdentifier)
+          .creationTime(Instant.now().getEpochSecond())
+          .expiryTime(Instant.now().getEpochSecond() + 1)
```

Comment:
It's a default storm tracking cache; won't these always be in Grace Period?

Why not make the expiry time greater than the Grace period?

Or switch the cache type to multi-threaded?

Then we won't have to catch the interrupt exception, because we will never be sleeping.